std/experimental/allocator/gc_allocator.d

```diff
@@ -80,14 +80,14 @@ struct GCAllocator
     }
 
     /// Ditto
-    @system bool deallocate(void[] b) shared
+    @trusted bool deallocate(void[] b) shared nothrow pure
```

Comment:
Fair enough, but can we then create a `SafeGCAllocator` that overloads common operations like `makeArray` with the language garbage collector and stubs `deallocate`?